### PR TITLE
[REF] cleanup o-spreadsheet code a bit

### PR DESCRIPTION
**Pull request by hokolomopo**
## Description:

I ran JetBrain's code inspection on the o-spreadhseet code. It listed a bunch of issues that I fixed.  Probably some of these commits are kinda useless.

Task: [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class PivotSidePanelStore extends SpreadsheetStore {
   }
 
   deferUpdates(shouldDefer: boolean) {
-    if (shouldDefer === false && this.draft) {
+    if (!shouldDefer && this.draft) {
```
Comment (VincentSchippefilt) [inline src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts:132]: is it really equivalent ?

File: src/functions/helpers.ts [hunk 2]
```diff
@@ -212,7 +212,7 @@ export function toBoolean(data: FunctionResultObject | CellValue | undefined): b
         return false;
       }
     case "number":
-      return value ? true : false;
+      return !!value;
```
Comment (VincentSchippefilt) [inline src/functions/helpers.ts:215]: it is less readable... I would stay away from !! even if it is valid javascript

File: src/plugins/ui_core_views/cell_evaluation/evaluator.ts [hunk 3]
```diff
@@ -456,7 +456,7 @@ export class Evaluator {
 
   private checkCollision(formulaPosition: CellPosition): (i: number, j: number) => void {
     const { sheetId, col, row } = formulaPosition;
-    const checkCollision = (i: number, j: number) => {
+    return (i: number, j: number) => {
```
Comment (VincentSchippefilt) [inline src/plugins/ui_core_views/cell_evaluation/evaluator.ts:459]: don't we loose the function name when inspecting the callstack ?

File: src/plugins/ui_stateful/sheetview.ts [hunk 4]
```diff
@@ -252,9 +252,6 @@ export class SheetViewPlugin extends UIPlugin {
       case "UNFREEZE_COLUMNS_ROWS":
         this.resetViewports(this.getters.getActiveSheetId());
         break;
-      case "DELETE_SHEET":
-        this.sheetsWithDirtyViewports.delete(cmd.sheetId);
-        break;
```
Comment (VincentSchippefilt) [inline src/plugins/ui_stateful/sheetview.ts:257]: why ?

File: src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts (hunk 1 above)
Comment (hokolomopo) [reply src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts:132]: It should be 🤔 I'll jsute revert it, that's true that it's less readable

File: src/functions/helpers.ts (hunk 2 above)
Comment (hokolomopo) [reply src/functions/helpers.ts:215]: You think ? I think it was Remi that said that it's better than Boolean(value). And I think both are better than `value ? true : false`. But I do'nt care that much, I'll remove the `!!` if you do'nt like them.

File: src/plugins/ui_core_views/cell_evaluation/evaluator.ts (hunk 3 above)
Comment (hokolomopo) [reply src/plugins/ui_core_views/cell_evaluation/evaluator.ts:459]: That's true, I didn't think arrow function could be named in the callstack, even if they were assigned to a variable. I'll change it back.

File: src/plugins/ui_stateful/sheetview.ts (hunk 4 above)
Comment (hokolomopo) [reply src/plugins/ui_stateful/sheetview.ts:257]: As the commit message says, the switch contained the `case "DELETE_SHEET"` twice. The second was thus never used.

**Comment by VincentSchippefilt** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**